exact.go

```diff
@@ -0,0 +1,106 @@
+// Package exact implements common math for rational numbers.
+// You must use this package if you need to avoid floating point
+// roudoff errors.
+package exact
+
+type (
+	// Frac is a fraction
+	Frac struct {
+		P int64 // P is the numerator
+		Q int64 // Q is the denominator
+	}
+)
+
+// Add two rationale numbers a and b.
+func Add(a, b Frac) Frac {
+	return Frac{
+		P: a.P*b.Q + b.P*a.Q,
+		Q: a.Q * b.Q,
+	}
+}
+
+// Sub subtract the rational numbers a and b.
+func Sub(a, b Frac) Frac {
+	return Add(a, Frac{
+		P: -b.P,
+		Q: b.Q,
+	})
+}
+
+// Mul multiplies a and b.
+func Mul(a, b Frac) Frac {
+	return Frac{
+		P: a.P * b.P,
+		Q: a.Q * b.Q,
+	}
+}
+
+// Div divides a and b
+func Div(a, b Frac) Frac {
+	return Mul(a, Frac{
+		P: b.Q,
+		Q: b.P,
+	})
+}
+
+// gcd is the greatest common dividor
+func gcd(a, b int64) int64 {
+	if b == 0 {
+		return a
+	}
+	return gcd(b, a%b)
+}
+
+// Simplify the fraction
+func Simplify(a Frac) Frac {
+	cd := gcd(a.P, a.Q) // common divisor
+	return Frac{
+		P: a.P / cd,
+		Q: a.Q / cd,
+	}
+}
+
+// Cmp compares if a equals b.
+func Cmp(a, b Frac) bool {
+	as := Simplify(a)
+	bs := Simplify(b)
+	return as.P == bs.P &&
+		as.Q == bs.Q
+}
+
+func reverse(s []rune) {
+	if len(s) < 2 {
+		return
+	}
+	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
+		s[i], s[j] = s[j], s[i]
+	}
+}
+
+// integer to []rune
+func itors(i int64, runeBase rune) []rune {
```

Comment:
I have no idea what this function does o.O (even with the docs, it seems like something important is going on but for someone stupid as me it is pretty hard to get it).

Reply:
The doc is missing a fundamental information, sorry. 
The idea is to convert a integer (like 12345) to an array of runes having `runeBase` as base unicode number.

The remaining of the division by 10 gives the unit number of the decimal number (the last character) and then dividing by 10 at each iteration we get all numbers from 0 to 9 that represents the original decimal number but with reversed order.

The runeBase is used to represent the number in a specific unicode class, in this case I was using the superscripts and subscripts to, for example, represent the fraction 1/2 as ½...
It seemed interesting at the time, but soon I realized it was a hell to debug, as copy-paste of the numbers to a calculator doesn't works properly..
Then I abandoned this idea in the last PR.

Reply:
Ow I see =D